lib/jxl/dec_xyb.cc

```diff
@@ -250,6 +251,8 @@ Status OutputEncodingInfo::MaybeSetColorEncoding(
 
 Status OutputEncodingInfo::SetColorEncoding(const ColorEncoding& c_desired) {
   color_encoding = c_desired;
+  linear_color_encoding = color_encoding;
+  linear_color_encoding.Tf().SetTransferFunction(TransferFunction::kLinear);
```

Comment:
Is this missing the call to `CreateICC()`?

Reply:
This call moved to here:

https://github.com/libjxl/libjxl/blob/6cc2150ecb9f3b219fc18bec592b6dacf34bc071/lib/jxl/dec_cache.cc#L242

because it was causing failures in the case where we don't set a CMS